roles/openshift_logging/templates/es.j2

```diff
@@ -95,6 +95,13 @@ spec:
               readOnly: true
             - name: elasticsearch-storage
               mountPath: /elasticsearch/persistent
+          readinessProbe:
+            exec:
+              command:
+              - "/usr/share/elasticsearch/probe/readiness.sh"
+            initialDelaySeconds: 5
+            timeoutSeconds: 5
```

Comment:
@lukas-vlcek, [here's the API type definition of a Probe](https://github.com/openshift/origin/blob/e37455a407f05ac584f5c462effc2b8c3626f39d/vendor/k8s.io/kubernetes/pkg/api/v1/types.go#L1300-L1326).

> If timeoutSeconds is not present, is any default values used for it instead?

Yes, 1 second. But, as I linked above, there's a long-standing issue that timeouts are not enforced for "exec" probes. So in practice, for the time being, setting this value has no effect.

> Does it mean that readiness probe works the same regardless timeoutSeconds being setup at all?

Yes, precisely. At least the Docker Remote API never supported timing out processes executed via the equivalent of a "docker exec ...". There is some ground work done to support this in other container backends, but AFAIK it is not a reality for most people/installations.

> If the command points to shell script which has internal timeout (specifically: curl --max-time ...) does it make sense to bother with probe timeoutSeconds at all? Can we just rely on the internal script timeout instead?

My recommendation would be setting a timeout on the command as you say, yes.